### PR TITLE
DAOS-1965 md: re-enable metadata add/remove test

### PR DESCRIPTION
**Pull request by kccain**
With this change the skipForTest decorator is removed from the server
metadata test function test_metadata_addremove(). Successful testing
is now possible due to PR #1635 that enabled rdb log compaction.

Skip-run_test: true
Test-tag: metadata_free_space
Test-tag-hw: metadata_free_space

Signed-off-by: Ken Cain <kenneth.c.cain@intel.com>

### CODE REVIEW

**Comment by kccain** [comment]
Functional test failures in builds 1 and 2 are of the type "failed to start server in normal mode" as noted in DAOS-4085. Artifacts from those runs have been uploaded to that JIRA ticket. I have run this test manually on a developer cluster successfully. For now, awaiting Functional_Hardware test run rather than try a 3rd time to get the functional test to run.

**Comment by kccain** [comment]
Build #4 functional test failure timeout - looking at the server log it is apparent that the DEBUG level of logging is completely inappropriate for a test such as this.

File: src/tests/ftest/server/metadata.py [hunk 1]
```diff
@@ -164,18 +163,18 @@ def test_metadata_addremove(self):
         Use Cases:
             ?
 
-        :avocado: tags=metadata,metadata_free_space,nvme,small
+        :avocado: tags=metadata,metadata_free_space,nvme,small,hw,full_regression
```
[the remark this answers is not in the document]
Comment (saurabhtandan) [reply src/tests/ftest/server/metadata.py]: You can avoid this by doing the following:
:avocado: tags=metadata,metadata_free_space,nvme,small,hw
:avocado: tags=full_regression

[the remark this answers is not in the document]
Comment (saurabhtandan) [reply src/tests/ftest/server/metadata.py]: Pardon the emoji. ':avocado:' automatically got translated to the emoji.

**Comment by kccain** [comment]
PR has passed functional_hardware testing and is ready to land once reviewed

**Pull request merged**